accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,188 @@
+
+# Motivation
+[motivation]: #motivation
+
+Reasons for use of Hub with Retail environment
+----------------------------------------------
+
+- scalability
+- disconnected operation
+
+
+Possible configurations
+-----------------------
+
+1. Full
+```
+    Terminal -> Branch Server -> Peripheral Server -> Hub
+```
+
+2. Combined SUMA and Branch server 
+```
+    Terminal -> Branch + Peripheral Server Combined -> Hub
+```
+
+3. Standard
+```
+    Terminal -> Branch Server -> SUMA Server
+```
+
+4. Single server
+```
+    Terminal -> Branch + SUMA Server Combined
+```
+
+
+Global Retail data
+------------------
+
+The following items are typically shared in whole organization, so they should be visible and 
+manageable via Hub.
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, list of packages
+  - HWTYPE configurations - group formula
+  
+Local data
+----------
+
+The following items do not need central management and can stay on Peripheral Server.
+
+  - Branch groups -  groups containing Branch Server and connected terminals
+  - Branch formulas - describe configuration of Branch services - network, pxe, dhcp, dns
+  - Branch PXE data - entries describing how to boot each terminal
+  
+# Detailed design
+[design]: #detailed-design
+
+Images + Image metadata
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
+  - Import Kiwi Image
+
+- Implement a tool that handles the use cases above
+
+
+HWTYPE configurations
+---------------------
+
+This is currently implemented as a formula assigned to a group with special name.
+
+Proposed solution:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+
+Alternative:
+Central creation of grups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+
+
+Combined SUMA and Branch server
```

Comment:
I am not sure how much is this relevant to Hub itself. I am missing a chapter discussing disconnected usage itself.

Also when talking about combined Uyuni and Branch server, it should be noted that we are talking about Uyuni taking up Branch responsibilities, but not becoming proxy itself, am I right? That means Uyuni peripheral server will provide necessary pxe, tftp, dhcp, dns and http services usually covered by Branch server proxy

Reply:
So if I understood correctly with the above discussion. Hub in retail scenario will still be an optional component and if an existing customer wants to keep the original setup with branch servers, they can keep it.

What I don't get is that if customer decide for hub and we decide to merge the branch server with peripheral server(susemanager). Will we have 2 different version of prephiral servers?

-  peripheral servers with possible branch server capabilities 
-  peripheral server(suse manager in current shape)

Reply:
> peripheral servers with possible branch server capabilities

Don't look at this like something special. It is just that this peripheral server will have enabled some retail formula (branch network, pxe, tftp and maybe dhcp and dns) and in some cases more than one network interface.
Once we figure out the problem with PXE entries I expect that solution to be used in all configuration.

Reply:
I would say it would be expected to have both.
Peripheral servers with branch server capabilities, which can be deployed in a store and manage terminal directly.
Also, a normal suse manager server which is responsible to manage proxy/branch servers deployed in several different stores.
And probably both coexisting in the same deployment infrastrucure.

Reply:
> I am not sure how much is this relevant to Hub itself.

In my understanding, it is in principle not relevant and could be tackled by a completely separate effort/RFC.

But once the capability of managing clients without a Proxy/RBS exists, then it's much easier to implement scenario 2 listed above ("just add a hub on top").

Am I on the right track here?

Reply:
Setup  without RBS is requested in the issue https://github.com/SUSE/spacewalk/issues/9594
I think it was in the original Hub requirements, but he link in the issue is now dead.